tests/conftest.py

```
@@ -0,0 +1,537 @@
+                    "Reference and result file must be of same file type!"
+                )
+            elif reference.suffix in [".vtk", ".vtu"]:
+                compare_vtk_files(reference, result, rtol, atol)
+            else:
+                raise NotImplementedError(
+                    f"Comparison is not yet implemented for {reference.suffix} files."
+                )
+
+        # String based comparison
+        else:
+
+            def get_string_to_compare(data):
+                """Get the string depending on the data type that will actually
+                be compared."""
+                if isinstance(data, str):
+                    return data
+                elif isinstance(data, Path):
+                    return get_string(data)
+                elif isinstance(data, InputFile):
+                    return data.get_string(**input_file_kwargs)
+                else:
+                    raise TypeError(f"Data type {type(data)} not implemented.")
+
+            # retrieve strings to compare
+            [reference_string, result_string] = [
+                get_string_to_compare(data) for data in [reference, result]
+            ]
+
+            # compare strings and handle non-matching strings
+            compare, error_message = compare_strings(
+                reference_string, result_string, rtol, atol, **kwargs
+            )
+            if not compare:
+                if isinstance(reference, Path):
+                    handle_unequal_strings(
+                        error_message,
+                        tmp_path,
+                        current_test_name,
+                        result_string,
+                        reference,
+                    )
+                raise AssertionError(error_message)
+
+    return _compare
+
+
+def compare_strings(
+    reference: str,
+    result: str,
+    rtol: Optional[float] = None,
+    atol: Optional[float] = None,
+    string_splitter: str = " ",
+) -> Tuple[bool, str]:
```

This weird return type is what I wanted to prevent - let's just raise an assertion error and catch it above? Then we do not need to return anything in this function as well and can simply reuse it

---

Yes, lets go back to your approach in this function, see my comment below.